core/services/periodicbackup/backup.go

```diff
@@ -0,0 +1,139 @@
+package periodicbackup
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/logger"
+)
+
+var (
+	fileName           = "db_backup.tar.gz"
+	minBackupFrequency = time.Minute
+)
+
+type backupResult struct {
+	size int64
+	path string
+}
+
+type (
+	DatabaseBackup interface {
+		Start() error
+		Close() error
+		RunBackupGracefully()
+	}
+
+	databaseBackup struct {
+		logger          *logger.Logger
+		databaseURL     url.URL
+		frequency       time.Duration
+		outputParentDir string
+		done            chan bool
+	}
+)
+
+func NewDatabaseBackup(frequency time.Duration, databaseURL url.URL, outputParentDir string, logger *logger.Logger) DatabaseBackup {
+	return &databaseBackup{
+		logger,
+		databaseURL,
+		frequency,
+		outputParentDir,
+		make(chan bool),
+	}
+}
+
+func (backup databaseBackup) Start() error {
+
+	ticker := time.NewTicker(backup.frequency)
+
+	go func() {
+		for {
+			select {
+			case <-backup.done:
+				ticker.Stop()
+				return
+			case <-ticker.C:
```

Comment:
What happens if a backup is still running when the ticker ticks again? Will it fail to remove the tempfile?

Reply:
no, the temp file has a different name each time, so it should be ok